File: 0202-happy-number/0202-happy-number.js
/**
 * @param {number} n
 * @return {boolean}
 */
var square= function(n){
    let res =0;
    while(n!=0){
        let rem = n % 10;
        res = res + rem*rem;
        n=Math.floor(n/10);
    }
    return res;
} 
var isHappy = function(n) {
    let slow= n;
    let  fast = n;
    while(1){
        slow = square(slow);
        fast = square(square(fast));
        if(fast==1){
            return true;
        }
        if(slow==fast){
            return false;
        }
    }
    return false;
    
};

File: 0202-happy-number/0202-happy-number.cpp
class Solution {
public:
    int sumsquare(int val) {
        int res = 0;
        while (val != 0) {
            int rem = val % 10;
            res = res + rem * rem;
            val = val / 10;
        }
        return res;
    }
    bool isHappy(int n) {
        bool hasloop= false;
        int slow = n;
        int  fast = n;
        if(sumsquare(n)==1) return true;
        while(fast!=1){
            slow = sumsquare(slow);
            fast = sumsquare(sumsquare(fast));
            if(slow==fast){
                hasloop =  true;
                break;
            }
        }

        if(hasloop) return false;

        return true;

    }
};